src/evict/evict_page.c

```diff
@@ -179,9 +179,17 @@ __evict_delete_ref(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 		 * something is busy, be sure that the page still ends up
 		 * marked deleted.
 		 */
-		if (ndeleted > pindex->entries / 10 && pindex->entries > 1 &&
-		    (ret = __wt_split_reverse(session, ref)) != EBUSY)
-			return (ret);
+		if (ndeleted > pindex->entries / 10 && pindex->entries > 1) {
+			if ((ret = __wt_split_reverse(session, ref)) == 0)
+				return (0);
```

Comment:
@keithbostic, I think this change has introduced a bug that will leave child refs locked.

Unless I am missing something, when we hit the `result_entries == 0` case in `__split_parent`, we will give up on the split but return 0.  The cleanup path in `__split_parent` used to flip the child ref to `WT_REF_DELETED` but this branch has removed that code, and I don't think the code here can know whether the ref was unlocked.

We could have that case return `EBUSY`, but then we would have to suppress the error message in that case because it is an expected condition.  One way to do that would be to pull the `__wt_err` call from `__split_parent` up into callers other than `__wt_split_reverse` (which I think just means `__split_insert` since the other caller has its own error message already.